app/controllers/checkouts_controller.rb

```diff
@@ -0,0 +1,10 @@
+class CheckoutsController < ApplicationController
+  before_action :require_login, :allow_anyone
```

Comment:
Ok, I see now... You have an after_filter that makes sure authorization has been checked and allow_anyone records that it has.